Parse/src/main/java/com/parse/ParseEncoder.java

```diff
@@ -120,6 +122,12 @@ public Object encode(Object object) {
         return ((ParseQuery.RelationConstraint) object).encode(this);
       }
 
+      if (object instanceof Pattern) {
+        JSONObject json = new JSONObject();
+        json.put("$regex", object.toString());
```

Comment:
I've updated the function to use KeyConstraints.

I keep using `whereContsinaAll` because was not restricted to `ParseObject`s. The docs were wrong.